satellite/gracefulexit/validation.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gracefulexit
+
+import (
+	"bytes"
+	"context"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/signing"
+	"storj.io/storj/uplink/eestream"
+)
+
+func validatePointer(ctx context.Context, pointer *pb.Pointer, incomplete *TransferQueueItem) (*pb.RemotePiece, error) {
+	remote := pointer.GetRemote()
+	nodeID := incomplete.NodeID
+
+	pieces := remote.GetRemotePieces()
+	var nodePiece *pb.RemotePiece
+	for _, piece := range pieces {
+		if piece.NodeId == nodeID && piece.PieceNum == incomplete.PieceNum {
+			nodePiece = piece
+		}
+	}
+
+	if nodePiece == nil {
+		return nil, Error.New("piece no longer held by node")
+	}
+
+	return nodePiece, nil
+}
+
+func validateRedundancyThreshold(ctx context.Context, pointer *pb.Pointer, redundancy eestream.RedundancyStrategy) bool {
```

Comment:
Does this need to be a separate function? I think it would be fine to have this logic in `calculatePieceSize`